src/coalesce-vue-vuetify3/src/components/input/c-datetime-picker.vue

```diff
@@ -584,6 +594,34 @@ function emitInput(value: Date | null) {
   }
 }
 
+function setToday() {
+  const now = new Date();
+  const today = startOfDay(now);
+
+  // Preserve existing time if it's set
+  if (internalValueZoned.value) {
+    const existingDate = internalValueZoned.value;
+
+    const newDate = set(today, {
+      hours: existingDate.getHours(),
+      minutes: existingDate.getMinutes(),
+      seconds: existingDate.getSeconds(),
+      milliseconds: existingDate.getMilliseconds(),
+    });
+
+    if (internalTimeZone.value) {
+      // Adjust for the specified time zone
+      modelValue.value = fromZonedTime(newDate, internalTimeZone.value);
+    } else {
+      // No time zone specified, use the new date as is
+      modelValue.value = newDate;
+    }
```

Comment:
Wasn't sure about this bit, it seems like it's possible we could run into a null time zone... Is it fine to just use the new Date() value in that case?